### DISCUSSION TITLE
New version of Catalog Items API, 2022-04-01

### DISCUSSION THREAD

**Discussion by therubioftw** [amzn/selling-partner-api-models#3332] [General] [answered]
First, congratulation on the new version, I know I'm not the only one who's been waiting for this release :)

While I appreciate all the new additions, there's one use-case I was sure was enabled, though it doesn't seem to be working as expected.

Let's say I want all the Items from one specific seller - shouldn't I be able to specify **marketplace** and **seller ID** with the search operation, and get back the list of those items?
Currently it forces me to use 'identifiers' or 'keywords', which seems like it defeats the purpose for some of the new search parameters...

**Comment by chapmanjw** [comment, marked as answer]
@therubioftw To help me better understand your use-case, are you looking for ASIN-data based on the listings a selling partner has or are you looking for SKU-data?

We are planning to launch a `searchListingsItems` operation on the Listings Items API in the future that would enable you to search and paginate through SKU-data for a selling partner. That will also include which ASIN the SKUs map to, allowing you to then query the `searchCatalogItems` operation for a list of ASINs or SKUs. 

For retrieving entire data sets (i.e. all ASIN data or all SKU data associated with the selling partner), bulk reports (in formats compatible with these APIs) may be a better solution. These are also on our roadmap for sometime in the future, though I don't have any firm timelines for these or the `searchListingsItems` operation at this time.

**Reply by therubioftw** [reply]
Thanks @chapmanjw!
Indeed, the `searchListingItems` operation you're describing would solve the issue.

While we are currently using a mix of reports to do this, the speed at with this happens leaves a lot to be desired. It's a very basic query that our clients do upon first onboarding to our platform, so waiting a few seconds makes sense, waiting a few minutes - doesn't...

I hope this clears up the use-case a bit, and the motivation behind it...

**Reply by chapmanjw** [reply]
What would be your ideal query pattern for this SKU data? Would you be qualifying the searches with some query parameters or would you want to paginate through everything? (The latter gets tricky as the an individual seller could have millions of items, which the APIs could not effectively page through in a timely manner).

**Reply by therubioftw** [reply]
That's a good point actually.
Our biggest clients so far have had thousands of potential SKUs, but if we only retrieve the ones that are actually actively selling first that would be a good start.
The way I tried to think of it, the easiest way to accommodate it would be to use `sellerID` as an `identifiersType` (so that the requirement for one is met).
If we could sort the results, by availability or rank, that would be great, but I'm being greedy :)

**Reply by chapmanjw** [reply]
@therubioftw I will be tagging this Github discussion to our design work for the `searchListingsItems` operation. If you have further suggestions, feel free to add them here. We will follow up once we pickup that work.

**Reply by therubioftw** [reply]
Amazing! Thanks again @chapmanjw, I really appreciate it, I don't take it for granted :)
